src/huggingface_hub/file_download.py

```diff
@@ -388,9 +388,9 @@ def _request_wrapper(
             `max_wait_time`.
         max_wait_time (`float`, *optional*, defaults to `2`):
             Maximum amount of time between two retries, in seconds.
-        timeout (`float`, *optional*, defaults to `10`):
+        timeout (`float`, *optional*):
             How many seconds to wait for the server to send data before
-            giving up which is passed to `requests.request`.
+            giving up which is passed to `requests.request`. Defaults to no timeouts.
```

Comment:
hmm there's still a timeout no? just controlled by the server...

I would default to 30s maybe

Reply:
i don't see the drawback of having a value on the client side honestly

Reply:
Ok, let's put 30s then. I don't see a good reason to put a value either but I'm fine with it

Reply:
Reverted to 30s timeout in https://github.com/huggingface/huggingface_hub/pull/1514/commits/8e16055a35d53cbde236f91c0bcb5b9ae75498e8

Reply:
code determinism if/when the server config changes

Reply:
and documentation